app/controllers/image_controller.rb

```diff
@@ -780,27 +780,21 @@ def row_changed?(row)
       row[:old_holder] != row[:new_holder]
   end
 
-  # Add license change records with raw SQL in order to use a single INSERT.
+  # Add license change records with a single insert to the db.
   # Otherwise updating would take too long for many (e.g. thousands) of images
   def update_licenses_history(images_to_update, old_holder, old_license_id)
     data = images_to_update.pluck(:id, :when)
+    values = data.each_with_object([]) do |datum, vals|
+      vals << { user_id: @user.id.to_s,
+                updated_at: Time.current,
```

Comment:
Is there a subtle difference here?
I think that `updated_at` be set to the time when Rails runs this method, rather than the time that the db inserts the record.
The difference is so small that I don't think it matters.
Also note that in Rails 7, we can get rid of this line.

Reply:
Could this be simplified and made easier to understand (I always struggle with inject-style constructs):

```suggestion
CopyrightChange.insert_all(images_to_update.map do |image|
  { user: @user,
    updated_at: Time.current,
    target: iamge,
    year: image.when.year,
    name: old_holder,
    license: old_license }
end)
```

Reply:
Oops, I meant for that suggestion to replace the entire body of the method.

Also, I agree that the tiny difference in updated_at is negligible.  This is all about protecting ourselves legally -- having an audit trail of what the copyright used to be, in case someone did something with the image back when the copyright was different.  A few milliseconds won't make a difference! :)